app/retail/emails.py

```diff
@@ -518,10 +518,50 @@ def render_funder_stale(github_username, days=60, time_as_str='a couple months')
     return response_html, response_txt
 
 
-def render_new_bounty(to_email, bounties, old_bounties, offset=3, trending_quests=[]):
+def render_new_bounty(to_email, bounties, old_bounties, offset=3, quest_of_the_day={}, upcoming_grant={}, upcoming_hackathon={}, latest_activities={}, from_date=date.today(), days_ago=7):
     from townsquare.utils import is_email_townsquare_enabled, is_there_an_action_available
-    email_style = (int(timezone.now().strftime("%-j")) + offset) % 24
+    from dashboard.models import Profile
+    from inbox.models import Notification
     sub = get_or_save_email_subscriber(to_email, 'internal')
+    if settings.DEBUG and False:
+        # for debugging email styles
+        email_style = 26
+    else:
+        email_style = (int(timezone.now().strftime("%-j")) + offset) % 24
+
+    # Get notifications count from the Profile.User of to_email
+    try:
+        profile = Profile.objects.filter(email__iexact=to_email).last()
+    except Profile.DoesNotExist:
+        pass
+
+    from_date = from_date + timedelta(days=1)
+    to_date = from_date - timedelta(days=days_ago)
+
+    try:
+        notifications_count = Notification.objects.filter(to_user=profile.user.id, is_read=False, created_on__range=[to_date, from_date]).count()
+    except Notification.DoesNotExist:
+        pass        
+
+    upcoming_events = []
+    if upcoming_grant:
+        upcoming_events.append({
+            'event': upcoming_grant,
+            'title': upcoming_grant.title,
+            'image_url': upcoming_grant.logo.url,
+            'url': upcoming_grant.url,
+            'date': upcoming_grant.next_clr_calc_date.strftime("%Y-%d-%m") if upcoming_grant.next_clr_calc_date else upcoming_grant.created_on.strftime("%Y-%d-%m")
```

Comment:
Why is the CLR calc date being used here?

Reply:
Since the **`Save The Date`** section is supposed to display upcoming dates, I thought `next_clr_calc_date` would be later than the time of creation of the email and `created_on` of the grant. So display `next_clr_calc_date` if it exists or creation date if it doesn't

Reply:
you should use grants.views.after_that_next_round_begin instead probably this is just the clr calc date